### DISCUSSION TITLE
ATP DB Refused to connect

### DISCUSSION THREAD

**Discussion by kumar-harshh** [oracle/python-oracledb#323] [Q&A]
Got error while connecting to the ATP DB using langchain loader. Not able to connect with the ATP DB without langchain either but I'm able to connect with this database in my SQLDeveloper. Removed the credentials for security purposes

Getting error in Python Program: 
DPY-6005: cannot connect to database (CONNECTION_ID=nRH8VLYW7j51qZvYDKt0Xw==).
DPY-6000: Listener refused connection. (Similar to ORA-12506)

from langchain_community.document_loaders import OracleAutonomousDatabaseLoader

SQL_QUERY = "select prod_id, time_id from sh.costs fetch first 5 rows only"

doc_loader = OracleAutonomousDatabaseLoader(
    query=SQL_QUERY,
    user='',
    password='',
    schema='',
    connection_string=''
)
doc = doc_loader.load()
print(doc)

**Comment by cjbj** [comment]
Read the [Connecting to Oracle Cloud Autonomous Databases](https://python-oracledb.readthedocs.io/en/latest/user_guide/connection_handling.html#connecting-to-oracle-cloud-autonomous-databases) documentation and follow what it says without using langchain.  Get that going first.

**Comment by anthony-tuininga** [comment]
The error `ORA-12506` indicates that your database requires mTLS or hasn't been configured correctly for one-way TLS. The documentation referenced by Chris should help you on your way there!

**Comment by kumar-harshh** [comment]
I've tried it either ways...with and without langchain.

However getting the same error. I also tried using TLS & mTLS..with and
without wallet credentials but the issue persists. I even tried allowing
all the request from my cloud console to this DB.

Suggestion Appreciated

**Reply by anthony-tuininga** [reply]
Did you supply the wallet location and password when using mTLS? That error implies that you don't have one-way TLS configured, or you are using the wrong port. There are different ports for TLS and mTLS configurations! It may be useful to check with another client like SQL*Plus.

**Reply by cjbj** [reply]
@kumar-harshh it all works for me. If you need more help, you'll have to share more details about what parameters you are passing and what values they are set to (you can redact your site info, as needed).
